test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/ModelBinderFactoryTest.cs

```diff
@@ -67,9 +67,46 @@ public void CreateBinder_CreatesNoOpBinder_WhenPropertyDoesntHaveABinder()
         }
 
         [Fact]
+        public void CreateBinder_CreatesNoOpBinder_WhenPropertyBindingIsNotAllowed()
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            metadataProvider
+                .ForProperty<Widget>(nameof(Widget.Id))
+                .BindingDetails(m => m.IsBindingAllowed = false);
+
+            var modelBinder = Mock.Of<IModelBinder>();
+
+            var options = new TestOptionsManager<MvcOptions>();
+            options.Value.ModelBinderProviders.Add(new TestModelBinderProvider(c =>
+            {
+                if (c.Metadata.ModelType == typeof(WidgetId))
+                {
+                    return modelBinder;
+                }
+
+                return null;
+            }));
+
+            var factory = new ModelBinderFactory(metadataProvider, options);
+
+            var context = new ModelBinderFactoryContext()
+            {
+                Metadata = metadataProvider.GetMetadataForProperty(typeof(Widget), nameof(Widget.Id)),
+            };
+
+            // Act
+            var result = factory.CreateBinder(context);
+
+            // Assert
```

Comment:
Suggest asserting that `result.GetType().FullName` contains "NoOp". Right now this test mainly demonstrates the `TestModelBinderProvider` isn't used.